src/chigame/users/views.py

```diff
@@ -120,3 +120,31 @@ def cancel_friend_invitation(request, pk):
     else:
         messages.error(request, "Something went wrong please try again later!")
     return redirect(reverse("users:user-profile", kwargs={"pk": request.user.pk}))
+
+
+@login_required
+def remove_friend(request, pk):
+    sender = User.objects.get(pk=request.user.id)
+    receiver = User.objects.get(pk=pk)
+    sender_profile = sender.profile
+    sender_profile.friends.remove(receiver)
+    messages.success(request, "Friend removed successfully.")
+    return redirect(reverse("users:user-profile", kwargs={"pk": request.user.pk}))
+
+@login_required
+def invite_to_game(request, pk, match_id):
+    sender = User.objects.get(pk=request.user.id)
+    receiver = User.objects.get(pk=pk)
+    match = User.objects.get(pk=match_id)  
+    GameInvitation.objects.create(sender=sender, receiver=receiver, group=group, match=match)
+    messages.success(request, "Invitation to game sent successfully.")
+    return redirect(reverse("users:user-profile", kwargs={"pk": request.user.pk}))
+
+@login_required
+def invite_to_tournament(request, pk, tournament_id):
+    sender = User.objects.get(pk=request.user.id)
+    receiver = User.objects.get(pk=pk)
+    tournament = User.objects.get(pk=tournament_id) 
```

Comment:
Same thing here where it should be Tournament.obkects instead of User.objects , as the User model doesn't have a tournament_id